runtime/common/src/assigned_slots.rs

```diff
@@ -0,0 +1,1280 @@
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This pallet allows to assign permanent (long-lived) or temporary
+//! (short-lived) parachain slots to paras, leveraging the existing
+//! parachain slot lease mechanism. Temporary slots are given turns
+//! in a fair (though best-effort) manner.
+//! The dispatchables must be called from the configured origin
+//! (typically `Sudo` or a governance origin).
+//! This pallet should not be used on a production relay chain,
+//! only on a test relay chain (e.g. Rococo).
+
+use crate::{
+	slots::{self, Pallet as Slots, WeightInfo},
+	traits::{LeaseError, Leaser, Registrar},
+};
+use frame_support::{pallet_prelude::*, traits::Currency};
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+use parity_scale_codec::Encode;
+use primitives::v1::Id as ParaId;
+use runtime_parachains::{
+	configuration,
+	paras::{self},
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{One, Saturating, Zero},
+	ArithmeticError,
+};
+use sp_std::prelude::*;
+
+/// Lease period an assigned slot should start from (current, or next one).
+#[derive(Encode, Decode, Clone, Copy, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum SlotLeasePeriodStart {
+	Current,
+	Next,
+}
+
+/// Information about a temporary parachain slot.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, RuntimeDebug, TypeInfo)]
+pub struct ParachainTemporarySlot<AccountId, LeasePeriod> {
+	/// Manager account of the para.
+	pub manager: AccountId,
+	/// Lease period the parachain slot should ideally start from,
+	/// As slot are allocated in a best-effort manner, this could be later,
+	/// but not earlier than the specified period.
+	pub period_begin: LeasePeriod,
+	/// Number of lease period the slot lease will last.
+	/// This is set to the value configured in `TemporarySlotLeasePeriodLength`.
+	pub period_count: LeasePeriod,
+	/// Last lease period this slot had a turn in (incl. current).
+	/// This is set to the beginning period of a slot.
+	pub last_lease: Option<LeasePeriod>,
+	/// Number of leases this temporary slot had (incl. current).
+	pub lease_count: u32,
+}
+
+type BalanceOf<T> = <<<T as Config>::Leaser as Leaser<<T as frame_system::Config>::BlockNumber>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+type LeasePeriodOf<T> =
+	<<T as Config>::Leaser as Leaser<<T as frame_system::Config>::BlockNumber>>::LeasePeriod;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	#[pallet::disable_frame_system_supertrait_check]
```

Comment:
why though?

Reply:
AFAIU the `runtime_parachains::configuration`'s Config trait already derives from frame_system's Config trait, so there is no need to derive explicitly from it again.